telemetry_package/main.py

```diff
@@ -1,33 +1,145 @@
 
+DEFAULT_UNIX_SOCKET_PATH = "/tmp/thi_drone"
 
 class TelemetryNode(CommonNode):
-    """A class that represents a telemetry node.
-    """
 
-    def __init__(self, id: str):
-        """Creates a new TelemetryNode.
+    def __init__(self):
+        super().__init__('telemetry_node')
+        qos_profile = QoSProfile( 
+            history=QoSHistoryPolicy.KEEP_LAST,
+            depth=1,
+            reliability=QoSReliabilityPolicy.BEST_EFFORT,
+            durability=QoSDurabilityPolicy.VOLATILE, 
+            liveliness = QoSLivelinessPolicy.AUTOMATIC
+        )
 
-        Args:
-            id (str): Unique node id
-        """
+        if os.path.exists(DEFAULT_UNIX_SOCKET_PATH): 
+            print('Socket already exists.')
+            raise FileExistsError
+            
+        self.server_sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
 
-        super().__init__(id)
+        try:
+            self.server_sock.bind(DEFAULT_UNIX_SOCKET_PATH)
+            self.server_sock.listen(1)
+
+            # Accept the connection
+            self.client_sock, _ = self.server_sock.accept()
+            print('Connection accepted. Creating subscription')
+
+            # After connecting, create two subscriptions: one to the '/rosout' topic and one to the 'control' topic
+            self.rosout_subscription = self.create_subscription(
+                Log,
+                '/rosout',
+                self.rosout_callback, 
+                qos_profile = qos_profile)
+            
+            self.rosout_subscription
+            print('Created subscription to "/rosout".')
+
+            # Commenting this out for test purposes
+            # self.control_subscription = self.create_subscription(
+            #     Log,
+            #     'control',
+            #     self.control_subscription_callback, 
+            #     qos_profile = qos_profile)
+
+            # self.control_subscription 
+            print('Created subscription to "control".')
+
+
+            self.heartbeat_subscription = self.create_subscription(
+               Log,
+               'heartbeat', 
+                qos_profile = qos_profile)
+            
+            self.heartbeat_subscription
+            print('created subscription to "heartbeat".')
+
+            
+
+        except Exception as e:
+            print(f'Error occurred: {e}')
+            self.server_sock.close()
+            os.remove(DEFAULT_UNIX_SOCKET_PATH)
+        
+
+    '''
+    @brief This function sanitizes incoming messages so that they do not collide with JSON format guidelines
+
+    @param msg message to sanitize
+
+    @return str
+    '''
+    def sanitize_message(self, msg : str)->str:
+        try: 
+            msg = re.sub("[\{\}:']", "", msg) 
+            return msg 
+        except Exception as e: 
+            print(f'Error occurred during string sanitization: {e}')
+
+
```

Comment:
```suggestion
```
You don't need to do sanitation, see [https://docs.python.org/3/library/json.html#json.dump](https://docs.python.org/3/library/json.html#json.dump)